pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -519,7 +521,11 @@ func (s *DiscoveryServer) initProxy(node *core.Node) (*model.Proxy, error) {
 	// This is not preferable as only the connected Pilot is aware of this proxies location, but it
 	// can still help provide some client-side Envoy context when load balancing based on location.
 	if util.IsLocalityEmpty(proxy.Locality) {
-		proxy.Locality = node.Locality
+		proxy.Locality = &corev3.Locality{
```

Comment:
Does this cause any issue if we use this `Locality` information in Listener/Routes?

Reply:
We don't use it in listener/routes. I don't think there is any fields there that take it in, but if we did need it we can always convert back and forth easily. LDS and RDS will be on v3 shortly after this PR lands though